event-handlers/src/main/java/no/sikt/nva/nvi/events/evaluator/EvaluatorService.java

```diff
@@ -136,15 +126,27 @@ private boolean isNonCandidate(PublicationDto publication) {
   }
 
   private boolean isPublished(PublicationDto publication) {
-    return nonNull(publication.status()) && "published".equalsIgnoreCase(publication.status());
+    return publication.status() != null && "published".equalsIgnoreCase(publication.status());
   }
 
-  private boolean isApplicableInPeriod(NviPeriod targetPeriod, Candidate candidate) {
-    var candidatePeriod = candidate.getPeriod();
-    if (isNull(candidatePeriod) || isNull(candidatePeriod.id())) {
+  private boolean canEvaluateInPeriod(
+      Optional<NviPeriod> optionalPeriod, Optional<Candidate> optionalCandidate) {
+    if (optionalPeriod.isEmpty()) {
```

Comment:
Consuming Optional in a method is a code smell. Optional was designed to be a return type, not a parameter. 
It signals `this method might not return a value`, not `pass me a maybe-value`.